src/Imgix/UrlBuilder.cs

```diff
@@ -77,6 +101,52 @@ private String GenerateUrl(String domain, String path, Dictionary<String, String
             return String.Format("{0}://{1}/{2}{3}", scheme, domain, path, localParams.Any() ? "?" + GenerateUrlStringFromDict(localParams) : String.Empty);
         }
 
+        private String GenerateSrcSetDPR(String domain, String path, Dictionary<String, String> parameters)
+        {
+            String srcset = "";
+            int[] targetRatios = { 1, 2, 3, 4, 5 };
+
+            foreach(int ratio in targetRatios)
+            {
+                parameters["dpr"] = ratio.ToString();
+                parameters.Remove("ixlib");
```

Comment:
@zacman85 Looks like the way we're adding `ixlib` (L57) is causing a collision error here unless we "remove" it between calls to `BuildUrl`. Curious if you think this strategy is preferable over say, changing L57 to use `parameters["ixlib"] = ...`, which will avoid the same issue but requires touching on the original `BuildUrl` implementation.

Reply:
In general, I do not like to have to remove things from a list like this, when something is being built up, unless it is unavoidable. I don't know enough about the codebase to suggest one way over the other as I don't know the consequences of a refactor here. I would leave it to your judgement.

Reply:
The immediate issue here is that `BuildUrl` indiscriminately `Add`s an `ixlib` parameter if the flag has been set. The real issues are more nuanced...

Nonetheless, `Remove` can be removed from the codebase if we just check if the `parameters`(the dictionary/hash-map that is passed) already contain the `ixlib` key-value pair. If so, continue; otherwise, if the flag is set, but no `ixlib` param is present, `Add` it.

Reply:
I would also go a bit further than this:

>In general, I do not like to have to remove things from a list like this, when something is being built up, unless it is unavoidable.

We're at **the** build step. All configuration should have already stopped by this point.